src/components/image-select.tsx

```diff
@@ -50,23 +51,20 @@ class ImageSelect extends React.Component<GridModalProps, GridModalState> {
       return;
     }
 
-    const data = event.data;
+    const postMessageService = new IframePostMessageService(event)
 
-    if (!data) {
+    if(!postMessageService.isValid) {
       return;
     }
 
-    if (!this.validMessage(data)) {
-      return;
-    }
+    const imageUrl: URL = postMessageService.highestQualityImageURL!;
 
-    const imageUrl = event.data.crop.data.master.secureUrl;
     this.setState({
-      imageId: event.data.image.data.id as string
+      imageId: postMessageService.imageId!
     });
 
     this.closeModal();
-    this.props.updateImageUrl(imageUrl);
+    this.props.updateImageUrl(imageUrl.toString());
     this.props.updateOriginalImageData(event.data.image);
```

Comment:
I think you may have missed one usage of the event object which can be replaced.
```suggestion
    this.props.updateOriginalImageData(postMessageService.data.image);
```

Reply:
Interesting, grid-client doesn't currently expose this data on the [`IframePostMessageService`](https://github.com/guardian/grid-client/blob/241f9464ac81684c2e7fd642a9d4270feca44a2c/src/services/iframe-post-message.ts). That is, `IframePostMessageService`'s current responsibility is to validate a payload and abstract away the logic of choosing the best asset from a crop.

What does `updateOriginalImageData` do with this data? AFAICT `event.data.image` is typed as `object`, I think introducing grid-client's more [complex type](https://github.com/guardian/grid-client/blob/241f9464ac81684c2e7fd642a9d4270feca44a2c/src/types/iframe-post-message.ts) here would be too much for this PR and we can be happy that the shape of `event.data.image` is correct here as we've previously validated with `postMessageService.isValid`. WDYT?

Reply:
I misunderstood this and I can see that `data` is protected. It would be nice to have some type-safety around this data, it is used exclusively for metadata and usage righters when uploading an image to the grid. This is something for another PR.

Reply:
Ah ok. I was planning on integrating grid-client's types in a follow up PR, so will take a look at this then.